tvm/src/pass/stream_inference.cc

```diff
@@ -3,31 +3,942 @@
  * \file stream_inference.cc
  * \brief mutate ir for scheduling streaming ops
  */
+#include <arithmetic/Substitute.h>
+#include <ir/IREquality.h>
 #include <tvm/ir.h>
 #include <tvm/ir_mutator.h>
 #include <tvm/ir_pass.h>
 #include <tvm/ir_visitor.h>
+#include <tvm/operation.h>
 #include <unordered_map>
 #include "./ir_util.h"
 
 namespace TVM {
 namespace ir {
 
+using std::string;
+using std::unordered_map;
+using std::unordered_set;
+using std::vector;
+
+struct IoInfo {
+  DeviceType dev_type;
+  StorageType storage_type;
+  int mem_port{-1};
+  StreamType stream_type;
+  int channel_depth{-1};
+  int burst_len{-1};
+};
```

Comment:
I believe we have ArgsInfo define somewhere...

Reply:
Yep. right now I encode this arginfo to a vector and then restore it back to the struct in codegen.